### PR TITLE
CORS-3453: Skip building libvirt terraform provider

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

**Comment by r4f4** [comment]
If we don't remove the provider, then this PR has to include changes to the baremetal-installer as well.

**Comment by sadasu** [comment]
> If we don't remove the provider, then this PR has to include changes to the baremetal-installer as well.

Done for baremetal and openstack too.

File: hack/build.sh [hunk 1]
```diff
@@ -42,9 +42,16 @@ check_module_changes() {
 # Build terraform and providers only if needed
 build_terraform_and_providers() {
 	TARGET_OS_ARCH=$(go env GOOS)_$(go env GOARCH)
+	# shellcheck disable=SC2039
+	SKIP_TERRAFORM_PROVIDERS="${SKIP_TERRAFORM_PROVIDERS:"none"}"
 	bindir="${PWD}/terraform/bin/${TARGET_OS_ARCH}"
 	find "${PWD}/terraform/providers/" -maxdepth 1 -mindepth 1 -type d | while read -r dir; do
 		provider="$(basename "${dir}")"
+		# shellcheck disable=SC2039
+		if echo "${SKIP_TERRAFORM_PROVIDERS[@]}" | grep "$provider"; then
```
Comment (r4f4) [inline hack/build.sh:51]: I think disabling the shellcheck here is not correct. It's actually warning about a potential issue: sometimes `SKIP_TERRAFORM_PROVIDERS` is an array and others it's a string.

**Review by r4f4 — CHANGES_REQUESTED** [review]

File: hack/build.sh (hunk 1 above)
Comment (sadasu) [reply hack/build.sh]: Keeping it a string throughout.

File: terraform/Makefile [hunk 2]
```diff
@@ -1,11 +1,13 @@
 TARGET_OS_ARCH:=$(shell go env GOOS)_$(shell go env GOARCH)
+UNSUPPORTED_PROVIDERS:=$(shell echo $SKIP_TERRAFORM_PROVIDERS)
 
 TFSUBDIRS:=	$(foreach DIR,$(shell find providers -maxdepth 1 -mindepth 1 -type d),$(subst providers/,,$(DIR)))
+SUPPORTEDTFSUBDIRS:= $(patsubst $(UNSUPPORTED_PROVIDERS),,$(TFSUBDIRS))
```
Comment (zaneb) [inline terraform/Makefile:5]:
```suggestion
SUPPORTEDTFSUBDIRS:= $(filter-out $(UNSUPPORTED_PROVIDERS),$(TFSUBDIRS))
```

using `patsubst` assumes that there is only ever 1 unsupported provider I think

File: terraform/Makefile [hunk 3]
```diff
@@ -1,11 +1,13 @@
 TARGET_OS_ARCH:=$(shell go env GOOS)_$(shell go env GOARCH)
+UNSUPPORTED_PROVIDERS:=$(shell echo $SKIP_TERRAFORM_PROVIDERS)
```
Comment (zaneb) [inline terraform/Makefile:2]:
```suggestion
UNSUPPORTED_PROVIDERS:=$(SKIP_TERRAFORM_PROVIDERS)
```

no need for shell here

**Review by zaneb — COMMENTED** [review]
I'm happy to either do this and then remove the provider, or just to remove the provider if nobody needs it in 4.16 any more. So approving both.
/approve

File: terraform/Makefile (hunk 2 above)
Comment (sadasu) [reply terraform/Makefile:5]: patsubst does replace multiple occurrences of a pattern. In this case, both work. I have replaced it with filter-out.

**Comment by pierreprinetti** [comment]
LGTM for OpenStack: the CI gives a good signal.
We have an unrelated issue with a permafailing load-balancer test, but that happens late enough in the job that I can say that the rest is OK.

**Comment by sadasu** [comment]
Don't need this step since we were ready to remove libvirt as a supported platform altogether.

**Pull request closed without merging**